### ISSUE TITLE
_ctypes.COMError when operating on document which is not in focus in Photoshop

### ISSUE THREAD

**Issue by blunderedbishop** [loonghao/photoshop-python-api#100]
Working on a document object which is not in focus in the Photoshop application throws this error
```
Traceback (most recent call last):
  File "/test.py", line 15, in <module>
    l.duplicate()
  File "\photoshop\api\_artlayer.py", line 463, in duplicate
    self.app.duplicate(relativeObject, insertionLocation)
  File "\comtypes\client\lazybind.py", line 182, in caller
    return self._comobj._invoke(descr.memid, descr.invkind, 0, *args)
  File "\comtypes\automation.py", line 729, in _invoke
    dp, var, None, argerr)
_ctypes.COMError: (-2147213312, None, (None, None, None, 0, None))
```

```
#test.py
import photoshop.api as ps

app = ps.Application()

for doc in app.documents:
    if len(doc.layers) > 1:
        print('Skipping', doc.name)
    else:
        for l in doc.layers:
            l.duplicate() # Throws error as above if doc is not in focus in photoshop app
```

 - OS: W10
 - Photoshop Version: Photoshop-2020
 - Python Version: 3.8

**Comment by blunderedbishop** [comment]
It seems necessary to set the active document before operating on it, like so:
```
# Import local modules
import photoshop.api as ps


app = ps.Application()

for doc in app.documents:
    app.activeDocument = doc #Sets the active document
    for l in doc.layers:
        l.duplicate()
        break
```

This solves the problem. Is there a better way to do this?

**Comment by blunderedbishop** [comment]
Also the contract method of the selection object is missing the argument by.

**Comment by loonghao** [comment]
> It seems necessary to set the active document before operating on it, like so:
> 
> ```
> # Import local modules
> import photoshop.api as ps
> 
> 
> app = ps.Application()
> 
> for doc in app.documents:
>     app.activeDocument = doc #Sets the active document
>     for l in doc.layers:
>         l.duplicate()
>         break
> ```
> 
> This solves the problem. Is there a better way to do this?

We can improve here
https://github.com/loonghao/photoshop-python-api/blob/master/photoshop/api/_documents.py#L64

**Comment by loonghao** [comment]
> Also the contract method of the selection object is missing the argument by.

Thank you for your report, 
I'll fix them over the weekend

**Comment by loonghao** [comment]
Hi @ilharp I have released a new version `0.17.1`

Cheers,

**Comment by loonghao** [comment]
@all-contributors please add @blunderedbishop for Bug reports.
